modules/programs/ne.nix

```
@@ -0,0 +1,84 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.programs.ne;
+
+  autoPrefFiles = let
+    autoprefs = if cfg.defaultPreferences != "" then
+      cfg.automaticPreferences // {
+        ".default" = cfg.defaultPreferences;
+      } # include defaultPreferences as .default#ap
+    else
+      cfg.automaticPreferences;
+    keys = builtins.attrNames autoprefs;
+    gen = key: {
+      name = ".ne/${key}#ap"; # file path
+      value = {
+        text = builtins.getAttr key autoprefs;
+      }; # content, in the expected "path".text format
+    };
+  in listToAttrs (map gen keys); # assembling these into a regular record
+in {
+  meta.maintainers = [ hm.maintainers.cwyc ];
+
+  options.programs.ne = {
+    enable = mkEnableOption "ne";
+
+    keybindings = mkOption {
+      type = types.lines;
+      default = "";
+      example = ''
+        KEY 7f BS
+        SEQ "\x1b[1;5D" 7f
+      '';
+      description = ''
+        Keybinding file for ne
+      '';
+    };
+    defaultPreferences = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Default preferences for ne </para><para>
+        Equivalent to programs.ne.automaticPreferences.".default"
+      '';
+    };
+    automaticPreferences = mkOption {
+      type = types.attrsOf types.str;
+      default = { };
+      example = {
+        "nix" = ''
+          TAB 0
+          TS 2
+        '';
+        "js" = ''
+          TS 4
+        '';
+      };
+      description = ''
+        Automatic preferences files for ne
+      '';
+    };
+    menus = mkOption {
+      type = types.lines;
+      default = "";
+      description = "Menu configuration file for ne";
+    };
+    virtualExtensions = mkOption {
+      type = types.lines;
+      default = "";
+      description = "Virtual extensions configuration file for ne";
+    };
+  };
+
+  config = mkIf cfg.enable ({
```

The parenthesis is unnecessary
```suggestion
  config = mkIf cfg.enable {
```